hibernate-core/src/main/java/org/hibernate/resource/jdbc/internal/LogicalConnectionProvidedImpl.java

```diff
@@ -28,6 +28,13 @@
 	private final boolean initiallyAutoCommit;
 	private boolean closed;
 
+	/**
+	 * This constructor should only be used when making a shareable connection or for testing.
+	 *
+	 * @param providedConnection - the provided connection.
+	 *
+	 * @see #makeShareableCopy()
+	 */
```

Comment:
AFAICT, only tests and `LogicalConnectionProvidedImpl#makeShareableCopy` use this constructor.

`LogicalConnectionProvidedImpl#makeShareableCopy` does not appear to be used. Should it be deprecated?

Also, I'm not sure if the new `ResourceRegistryStandardImpl` should use the original `JdbcObserver`. If it should, then this class needs to be reworked.

Reply:
I heard back from @sebersole that `LogicalConnectionProvidedImpl#makeShareableCopy` can be deprecated.

Reply:
I'm going to remove this constructor, since the caller should provide the `ResourceRegistry`.